File: PY-OOP/Player.py
class Player:
    player = []
    currentTrack = "Something"
    changedTrack = ''
    change = False


    @classmethod
    def Add(cls, TrackName):
        Player.player.append(TrackName)
        return Player.player

    @classmethod
    def Play(cls):
        log = Player.change
        if log == False:
            ct = Player.player[len(Player.player)-1]
            Player.currentTrack = ct
            print(f'Playing {Player.currentTrack.title} by {Player.currentTrack.artist}')
        else:
            print(f'Playing {Player.changedTrack.title} by {Player.changedTrack.artist}')
            Player.change = False

    @classmethod
    def Prev(cls):
        Player.change = True
        arr = Player.player
        n = Player.currentTrack
        ind = arr.index(n)
        if  ind < 0:
            ind = len(arr)
        else:
            ct = Player.player[ind - 1]
            Player.changedTrack = ct
            Player.currentTrack = ct
                # return Player.changedTrack

    @classmethod
    def Next(cls):
        Player.change = True
        arr = Player.player
        n = Player.currentTrack
        ind = arr.index(n)
        if ind > len(arr):
            ind = 0
        else:
            ct = Player.player[ind + 1]
            Player.changedTrack = ct
            Player.currentTrack = ct

    @classmethod
    def printTracksInfo(cls):
        arr = Player.player
        a = 0
        for i in arr:
            a +=1
            print(f"№{a} - track: {i.title}, album: {i.album}, artist: {i.artist}")

    @classmethod
    def selectTrack(cls, num):
        arr = Player.player

        if num > len(arr):
            print("There is no that number of Track")
        else:
            Player.change = True
            ct = arr[num]
            Player.changedTrack = ct

File: OOP JS/Player.js
class Player{
    constructor(){

        this.playlist = {
            
        };
        this.currentTrack = 0;

    }

  
}

let player = new Player();

class Track{

    constructor(_artist, _title, _album){
        this.artist=_artist;
        this.title = _title;
        this.album = _album;

    }

}

Player.prototype.add = function(newTrack){
        let arrKeys = Object.keys(this.playlist);
        this.playlist[arrKeys.length] = newTrack;

}

Player.prototype.play = function(){

    let cur = this.currentTrack;
   let curTrack = this.playlist[cur];
//    let curArtist 
    console.log(`Now is playing ${curTrack.title} by ${curTrack.artist}`)
    //  console.log(cur);
}

Player.prototype.next = function(){
        let arrKeys = Object.keys(this.playlist);
        let num = this.currentTrack;
        if(num < arrKeys.length-1){
  this.currentTrack++;
       }
        else{
            this.currentTrack = 0;
       
            }
   
}

Player.prototype.previous = function(){
    let arrKeys = Object.keys(this.playlist);
    if(this.currentTrack > 0){
         this.currentTrack--;
    }else{
      return this.currentTrack =  arrKeys.length-1;
    }
  
}




var driveTrack = new Track('Incubus', 'Drive', 'Make Yourself');
var laBambaTrack = new Track('Ritchie Valens', 'La Bamba', 'La Bamba');

player.add(driveTrack);
player.add(laBambaTrack);
player.previous();
player.previous();
player.previous();

player.play();



